aicsimageio/aics_series.py

```diff
@@ -0,0 +1,388 @@
+    * This class may be a temporary stop gap until chunked image reading is added to the core `AICSImage` class and if
+    and when that occurs, this class will be deprecated.
+    * No metadata is checked for ensuring that the data in each image is consistent across them all. Example being,
+    channels in different indicies in different files across the series.
+    """
+
+    def __init__(
+        self,
+        images: Union[types.PathLike, Iterable[types.PathLike]],
+        series_dim: str
+    ):
+        """
+        Constructor for AICSSeries class intended for providing an interface for image reading of a list of path like
+        references.
+
+        Parameters
+        ----------
+        images: Union[types.PathLike, Iterable[types.PathLike]]
+            Either a pathlike value to a directory with images or the ordered iterable set of pathlike objects to read
+            from for the series.
+        series_dim: str
+            Which dimension to override with the iterable images.
+        """
+        # Check if provided a directory, if so, get the directory contents
+        if isinstance(images, (str, Path)):
+            # Expand and handle path
+            images = Path(images).expanduser().resolve(strict=True)
+
+            # Ensure that it is actually a directory and not a single file
+            if not images.is_dir():
+                raise NotADirectoryError(
+                    f"Provided a single file: {images}. Did you mean to use `AICSImage`?"
+                )
+
+            # Set images to the contents of the directory
+            images = sorted(images.iterdir())
+
+            # Drop any files that start with `.`
+            # TODO: Is this actually a valid operation to do for the user or way to specific?
+            # TODO: Different option, find the most common suffix and use it as the base?
+            predrop_count = len(images)
+            images = [img for img in images if img.name[0] != "."]
+            log.debug(f"Dropped {predrop_count - len(images)} files that began with a `.`.")
+
+        # At this point, images should have either been converted to a list of files if provided a directory
+        # Or was given an iterable of file like
+        if not isinstance(images, Iterable):
+            raise TypeError(
+                f"AICSSeries requires either a path to a directory or an iterable of files."
+            )
+
+        # Check that the iterable is more than one image
+        if len(images) <= 1:
+            raise ValueError(f"Provided a single file: {images}. Did you mean to use `AICSImage`?")
+
+        # Now check that everything in the iterable is file like
+        # This will also raise an error if any of the file like paths was not found
+        images = [Path(img).expanduser().resolve(strict=True) for img in images]
+
+        # Ensure that the series dim provided is part of the normal set and that only one dim is provided
+        series_dim = series_dim.upper()
```

Comment:
two lines down from here, you are checking for isinstance(str)... do you want to do that before calling .upper() ?